src/Compilers/CSharp/Test/Semantic/Semantics/GenericConstraintsTests.cs

```diff
@@ -3629,5 +3629,343 @@ public struct YourStruct<T> where T : unmanaged
+                    //         var size = sizeof(MyStruct<int>);
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "sizeof(MyStruct<int>)").WithArguments("unmanaged generic structs", "8.0").WithLocation(8, 20)
+                );
+        }
+
+        [Fact]
+        public void GenericImplicitStackallocRequiresCSharp8()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+
+    public static unsafe void Test()
+    {
+        var arr = stackalloc[] { new MyStruct<int>() };
+    }
+}
+";
+            CreateCompilation(code, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                    // (8,19): error CS8370: Feature 'unmanaged generic structs' is not available in C# 7.3. Please use language version 8.0 or greater.
+                    //         var arr = stackalloc[] { new MyStruct<int>() };
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "stackalloc[] { new MyStruct<int>() }").WithArguments("unmanaged generic structs", "8.0").WithLocation(8, 19)
+                );
+        }
+
+        [Fact]
+        public void GenericStackallocRequiresCSharp8()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+
+    public static unsafe void Test()
+    {
+        var arr = stackalloc MyStruct<int>[4];
+    }
+}
+";
+            CreateCompilation(code, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                    // (8,30): error CS8370: Feature 'unmanaged generic structs' is not available in C# 7.3. Please use language version 8.0 or greater.
+                    //         var arr = stackalloc MyStruct<int>[4];
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "MyStruct<int>").WithArguments("unmanaged generic structs", "8.0").WithLocation(8, 30)
+                );
+        }
+
+        [Fact]
+        public void GenericStructPointerFieldRequiresCSharp8()
+        {
+            var code = @"
+public struct MyStruct<T>
+{
+    public T field;
+}
+
+public unsafe struct OtherStruct
+{
+    public MyStruct<int>* ms;
+}
+";
+            CreateCompilation(code, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                    // (9,12): error CS8370: Feature 'unmanaged generic structs' is not available in C# 7.3. Please use language version 8.0 or greater.
+                    //     public MyStruct<int>* ms;
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_3, "MyStruct<int>*").WithArguments("unmanaged generic structs", "8.0").WithLocation(9, 12)
+                );
+        }
+
+        [Fact]
+        public void StructContainingTuple_Unmanaged_RequiresCSharp8()
```

Comment:
Should have a version of this where the types of the tuple are a generic parameter that is constrained to unmanaged.